## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 477589`
Commit: [`30b87dd66964b2b7876de346350e7bcad409d147`](https://github.com/NixOS/nixpkgs/commit/30b87dd66964b2b7876de346350e7bcad409d147) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/30b87dd66964b2b7876de346350e7bcad409d147..pull/477589/head))
Merge: [`3389fb0977649e19fd67766260ac721e97b24343`](https://github.com/NixOS/nixpkgs/commit/3389fb0977649e19fd67766260ac721e97b24343)

Logs: https://github.com/magnetophon/nixpkgs-review-gha/actions/runs/20766573794


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>ctagdrc</li>
    <li>dexed</li>
    <li>fire</li>
    <li>qdelay</li>
    <li>ripplerx</li>
    <li>sg-323</li>
    <li>sonobus</li>
    <li>vaporizer2</li>
    <li>zlcompressor</li>
    <li>zlequalizer</li>
    <li>zlsplitter</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>ctagdrc</li>
    <li>dexed</li>
    <li>fire</li>
    <li>qdelay</li>
    <li>ripplerx</li>
    <li>sg-323</li>
    <li>sonobus</li>
    <li>vaporizer2</li>
    <li>zlcompressor</li>
    <li>zlequalizer</li>
    <li>zlsplitter</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>dexed</li>
    <li>fire</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>qdelay</li>
    <li>sg-323</li>
    <li>zlcompressor</li>
    <li>zlequalizer</li>
    <li>zlsplitter</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>dexed</summary>
<pre>      |       ^
/nix/store/1jfvmacg17fwd9g5l040c5715q9487gr-libcxx-19.1.2+apple-sdk-15.5/include/c++/v1/__memory/unique_ptr.h:262:71: note: in instantiation of member function &#x27;std::unique_ptr&lt;NSAccessibilityElement&lt;NSAccessibility&gt;, juce::AccessibleObjCClassDeleter&gt;::reset&#x27; requested here
  262 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 ~unique_ptr() { reset(); }
      |                                                                       ^
/nix/build/nix-33328-250233649/source/libs/JUCE/modules/juce_gui_basics/native/accessibility/juce_Accessibility_mac.mm:38:14: note: in instantiation of member function &#x27;std::unique_ptr&lt;NSAccessibilityElement&lt;NSAccessibility&gt;, juce::AccessibleObjCClassDeleter&gt;::~unique_ptr&#x27; requested here
   38 |     explicit AccessibilityNativeImpl (AccessibilityHandler&amp; handler)
      |              ^
/nix/build/nix-33328-250233649/source/libs/JUCE/modules/juce_gui_basics/native/accessibility/juce_Accessibility_mac.mm:27:13: note: &#x27;juceFreeAccessibilityPlatformSpecificData&#x27; has been marked as being introduced in macOS 10.10 here, but the deployment target is macOS 10.9.0
   27 | static void juceFreeAccessibilityPlatformSpecificData (NSAccessibilityElement&lt;NSAccessibility&gt;*)  {}
      |             ^
/nix/build/nix-33328-250233649/source/libs/JUCE/modules/juce_gui_basics/native/accessibility/juce_AccessibilitySharedCode_mac.mm:35:9: note: enclose &#x27;juceFreeAccessibilityPlatformSpecificData&#x27; in an @available check to silence this warning
   35 |         juceFreeAccessibilityPlatformSpecificData (element);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~           
[ 55%] Building CXX object Source/CMakeFiles/Dexed.dir/__/libs/JUCE/modules/juce_events/juce_events.mm.o
[ 56%] Building CXX object Source/CMakeFiles/Dexed.dir/__/libs/JUCE/modules/juce_core/juce_core.mm.o
1 error generated.
make[2]: *** [Source/CMakeFiles/Dexed.dir/build.make:559: Source/CMakeFiles/Dexed.dir/__/libs/JUCE/modules/juce_gui_basics/juce_gui_basics.mm.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:497: Source/CMakeFiles/Dexed.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
<details>
<summary>fire</summary>
<pre>[100%] Linking CXX CFBundle shared module Fire_artefacts/Release/VST3/Fire.vst3/Contents/MacOS/Fire
warning: unhandled Platform key FamilyDisplayName
error: tool &#x27;codesign&#x27; not found
-- Replacing invalid signature with ad-hoc signature
warning: unhandled Platform key FamilyDisplayName
error: tool &#x27;codesign&#x27; not found
make[2]: *** [CMakeFiles/Fire_LV2.dir/build.make:265: Fire_artefacts/Release/LV2/Fire.lv2/libFire.so] Segmentation fault: 11
make[2]: *** Deleting file &#x27;Fire_artefacts/Release/LV2/Fire.lv2/libFire.so&#x27;
make[1]: *** [CMakeFiles/Makefile2:387: CMakeFiles/Fire_LV2.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[100%] Built target Fire_CLAP
warning: unhandled Platform key FamilyDisplayName
error: tool &#x27;codesign&#x27; not found
-- Replacing invalid signature with ad-hoc signature
warning: unhandled Platform key FamilyDisplayName
error: tool &#x27;codesign&#x27; not found
removing moduleinfo.json
creating /nix/build/nix-33328-250233650/source/build/Fire_artefacts/Release/VST3/Fire.vst3
[100%] Built target Fire_VST3
make: *** [Makefile:146: all] Error 2</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>dexed</li>
    <li>fire</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>qdelay</li>
    <li>sg-323</li>
    <li>zlcompressor</li>
    <li>zlequalizer</li>
    <li>zlsplitter</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>dexed</summary>
<pre>      |       ^
/nix/store/5mhwf4lq5sjl102jk850i6a18al4x0jp-libcxx-19.1.2+apple-sdk-15.5/include/c++/v1/__memory/unique_ptr.h:262:71: note: in instantiation of member function &#x27;std::unique_ptr&lt;NSAccessibilityElement&lt;NSAccessibility&gt;, juce::AccessibleObjCClassDeleter&gt;::reset&#x27; requested here
  262 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 ~unique_ptr() { reset(); }
      |                                                                       ^
/nix/build/nix-6213-2899467576/source/libs/JUCE/modules/juce_gui_basics/native/accessibility/juce_Accessibility_mac.mm:38:14: note: in instantiation of member function &#x27;std::unique_ptr&lt;NSAccessibilityElement&lt;NSAccessibility&gt;, juce::AccessibleObjCClassDeleter&gt;::~unique_ptr&#x27; requested here
   38 |     explicit AccessibilityNativeImpl (AccessibilityHandler&amp; handler)
      |              ^
/nix/build/nix-6213-2899467576/source/libs/JUCE/modules/juce_gui_basics/native/accessibility/juce_Accessibility_mac.mm:27:13: note: &#x27;juceFreeAccessibilityPlatformSpecificData&#x27; has been marked as being introduced in macOS 10.10 here, but the deployment target is macOS 10.9.0
   27 | static void juceFreeAccessibilityPlatformSpecificData (NSAccessibilityElement&lt;NSAccessibility&gt;*)  {}
      |             ^
/nix/build/nix-6213-2899467576/source/libs/JUCE/modules/juce_gui_basics/native/accessibility/juce_AccessibilitySharedCode_mac.mm:35:9: note: enclose &#x27;juceFreeAccessibilityPlatformSpecificData&#x27; in an @available check to silence this warning
   35 |         juceFreeAccessibilityPlatformSpecificData (element);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~           
[ 55%] Building CXX object Source/CMakeFiles/Dexed.dir/__/libs/JUCE/modules/juce_events/juce_events.mm.o
[ 56%] Building CXX object Source/CMakeFiles/Dexed.dir/__/libs/JUCE/modules/juce_core/juce_core.mm.o
1 error generated.
make[2]: *** [Source/CMakeFiles/Dexed.dir/build.make:559: Source/CMakeFiles/Dexed.dir/__/libs/JUCE/modules/juce_gui_basics/juce_gui_basics.mm.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:497: Source/CMakeFiles/Dexed.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
<details>
<summary>fire</summary>
<pre>ld: warning: object file (/nix/store/2cmn7azxqlji4m37hl0j0wkhc32k0rbx-catch2-3.11.0/lib/libCatch2.a(catch_xmlwriter.cpp.o)) was built for newer macOS version (14.0) than being linked (11.0)
ld: warning: object file (/nix/store/2cmn7azxqlji4m37hl0j0wkhc32k0rbx-catch2-3.11.0/lib/libCatch2.a(catch_matchers.cpp.o)) was built for newer macOS version (14.0) than being linked (11.0)
ld: warning: object file (/nix/store/2cmn7azxqlji4m37hl0j0wkhc32k0rbx-catch2-3.11.0/lib/libCatch2.a(catch_matchers_container_properties.cpp.o)) was built for newer macOS version (14.0) than being linked (11.0)
ld: warning: object file (/nix/store/2cmn7azxqlji4m37hl0j0wkhc32k0rbx-catch2-3.11.0/lib/libCatch2.a(catch_matchers_exception.cpp.o)) was built for newer macOS version (14.0) than being linked (11.0)
ld: warning: object file (/nix/store/2cmn7azxqlji4m37hl0j0wkhc32k0rbx-catch2-3.11.0/lib/libCatch2.a(catch_matchers_floating_point.cpp.o)) was built for newer macOS version (14.0) than being linked (11.0)
ld: warning: object file (/nix/store/2cmn7azxqlji4m37hl0j0wkhc32k0rbx-catch2-3.11.0/lib/libCatch2.a(catch_matchers_predicate.cpp.o)) was built for newer macOS version (14.0) than being linked (11.0)
ld: warning: object file (/nix/store/2cmn7azxqlji4m37hl0j0wkhc32k0rbx-catch2-3.11.0/lib/libCatch2.a(catch_matchers_quantifiers.cpp.o)) was built for newer macOS version (14.0) than being linked (11.0)
ld: warning: object file (/nix/store/2cmn7azxqlji4m37hl0j0wkhc32k0rbx-catch2-3.11.0/lib/libCatch2.a(catch_matchers_string.cpp.o)) was built for newer macOS version (14.0) than being linked (11.0)
ld: warning: object file (/nix/store/2cmn7azxqlji4m37hl0j0wkhc32k0rbx-catch2-3.11.0/lib/libCatch2.a(catch_matchers_templated.cpp.o)) was built for newer macOS version (14.0) than being linked (11.0)
ld: warning: object file (/nix/store/2cmn7azxqlji4m37hl0j0wkhc32k0rbx-catch2-3.11.0/lib/libCatch2.a(catch_matchers_impl.cpp.o)) was built for newer macOS version (14.0) than being linked (11.0)
warning: unhandled Platform key FamilyDisplayName
error: tool &#x27;codesign&#x27; not found
-- Replacing invalid signature with ad-hoc signature
warning: unhandled Platform key FamilyDisplayName
error: tool &#x27;codesign&#x27; not found
removing moduleinfo.json
creating /nix/build/nix-6213-2899467577/source/build/Fire_artefacts/Release/VST3/Fire.vst3
[ 99%] Built target Fire_VST3
[ 99%] Built target Benchmarks
make: *** [Makefile:146: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/ct/ctagdrc/package.nix
+++ b/pkgs/by-name/ct/ctagdrc/package.nix
@@ -48,6 +48,17 @@ stdenv.mkDerivation (finalAttrs: {
     libXtst
   ];
 
+  env.NIX_CFLAGS_COMPILE = toString [
+    # juce, compiled in this build as part of a Git submodule, uses `-flto` as
+    # a Link Time Optimization flag, and instructs the plugin compiled here to
+    # use this flag to. This breaks the build for us. Using _fat_ LTO allows
+    # successful linking while still providing LTO benefits. If our build of
+    # `juce` was used as a dependency, we could have patched that `-flto` line
+    # in our juce's source, but that is not possible because it is used as a
+    # Git Submodule.
+    "-ffat-lto-objects"
+  ];
+
   enableParallelBuilding = true;
 
   postPatch = ''
@@ -55,8 +66,6 @@ stdenv.mkDerivation (finalAttrs: {
           --replace-fail 'COPY_PLUGIN_AFTER_BUILD TRUE' 'COPY_PLUGIN_AFTER_BUILD FALSE'
     substituteInPlace CMakeLists.txt \
           --replace-fail 'include(cmake-include/CPM.cmake)' '# No CPM needed'
-    substituteInPlace CMakeLists.txt \
-          --replace-fail 'juce::juce_recommended_lto_flags' '# Not forcing LTO'
   '';
 
   installPhase = ''

--- a/pkgs/by-name/de/dexed/package.nix
+++ b/pkgs/by-name/de/dexed/package.nix
@@ -33,11 +33,6 @@ stdenv.mkDerivation (finalAttrs: {
 
     substituteInPlace Source/CMakeLists.txt \
       --replace-fail 'COPY_PLUGIN_AFTER_BUILD TRUE' 'COPY_PLUGIN_AFTER_BUILD FALSE'
-  ''
-  # LTO needs special setup on Linux
-  + lib.optionalString stdenv.hostPlatform.isLinux ''
-    substituteInPlace Source/CMakeLists.txt \
-      --replace-fail 'juce::juce_recommended_lto_flags' '# Not forcing LTO'
   '';
 
   strictDeps = true;
@@ -66,6 +61,17 @@ stdenv.mkDerivation (finalAttrs: {
     "-lXinerama"
     "-lXrandr"
     "-ljack"
+  ]);
+
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isLinux (toString [
+    # juce, compiled in this build as part of a Git submodule, uses `-flto` as
+    # a Link Time Optimization flag, and instructs the plugin compiled here to
+    # use this flag to. This breaks the build for us. Using _fat_ LTO allows
+    # successful linking while still providing LTO benefits. If our build of
+    # `juce` was used as a dependency, we could have patched that `-flto` line
+    # in our juce's source, but that is not possible because it is used as a
+    # Git Submodule.
+    "-ffat-lto-objects"
   ]);
 
   installPhase =

--- a/pkgs/by-name/fi/fire/package.nix
+++ b/pkgs/by-name/fi/fire/package.nix
@@ -108,11 +108,6 @@ stdenv.mkDerivation (finalAttrs: {
       # Regression tests require big sound files stored in LFS, skip them
       rm -v tests/RegressionTests.cpp
     ''
-    + lib.optionalString stdenv.hostPlatform.isLinux ''
-      # Remove hardcoded LTO flags: needs extra setup on Linux
-      substituteInPlace CMakeLists.txt \
-        --replace-fail 'juce::juce_recommended_lto_flags' '# Not forcing LTO'
-    ''
     + lib.optionalString (!finalAttrs.finalPackage.doCheck) ''
       substituteInPlace CMakeLists.txt \
         --replace-fail 'include(Tests)' '# Not building tests' \
@@ -143,6 +138,17 @@ stdenv.mkDerivation (finalAttrs: {
     (lib.cmakeBool "CPM_LOCAL_PACKAGES_ONLY" true)
     (lib.cmakeFeature "Catch2_SOURCE_DIR" "${catch2_3.src}")
   ];
+
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isLinux (toString [
+    # juce, compiled in this build as part of a Git submodule, uses `-flto` as
+    # a Link Time Optimization flag, and instructs the plugin compiled here to
+    # use this flag to. This breaks the build for us. Using _fat_ LTO allows
+    # successful linking while still providing LTO benefits. If our build of
+    # `juce` was used as a dependency, we could have patched that `-flto` line
+    # in our juce's source, but that is not possible because it is used as a
+    # Git Submodule.
+    "-ffat-lto-objects"
+  ]);
 
   installPhase = ''
     runHook preInstall

--- a/pkgs/by-name/pr/proteus/package.nix
+++ b/pkgs/by-name/pr/proteus/package.nix
@@ -50,7 +50,13 @@ stdenv.mkDerivation rec {
   env.NIX_CFLAGS_COMPILE = toString [
     # Support JACK output in the standalone application:
     "-DJUCE_JACK"
-    # Accommodate -flto:
+    # juce, compiled in this build as part of a Git submodule, uses `-flto` as
+    # a Link Time Optimization flag, and instructs the plugin compiled here to
+    # use this flag to. This breaks the build for us. Using _fat_ LTO allows
+    # successful linking while still providing LTO benefits. If our build of
+    # `juce` was used as a dependency, we could have patched that `-flto` line
+    # in our juce's source, but that is not possible because it is used as a
+    # Git Submodule.
     "-ffat-lto-objects"
   ];
 

--- a/pkgs/by-name/qd/qdelay/package.nix
+++ b/pkgs/by-name/qd/qdelay/package.nix
@@ -61,10 +61,16 @@ stdenv.mkDerivation (finalAttrs: {
     "-DCMAKE_OSX_ARCHITECTURES=${stdenv.hostPlatform.darwinArch}"
   ];
 
-  postPatch = ''
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'juce::juce_recommended_lto_flags' '# Not forcing LTO'
-  '';
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isLinux (toString [
+    # juce, compiled in this build as part of a Git submodule, uses `-flto` as
+    # a Link Time Optimization flag, and instructs the plugin compiled here to
+    # use this flag to. This breaks the build for us. Using _fat_ LTO allows
+    # successful linking while still providing LTO benefits. If our build of
+    # `juce` was used as a dependency, we could have patched that `-flto` line
+    # in our juce's source, but that is not possible because it is used as a
+    # Git Submodule.
+    "-ffat-lto-objects"
+  ]);
 
   installPhase = ''
     runHook preInstall

--- a/pkgs/by-name/ri/ripplerx/package.nix
+++ b/pkgs/by-name/ri/ripplerx/package.nix
@@ -61,14 +61,19 @@ stdenv.mkDerivation (finalAttrs: {
     "-lXrandr"
   ];
 
+  env.NIX_CFLAGS_COMPILE = toString [
+    # juce, compiled in this build as part of a Git submodule, uses `-flto` as
+    # a Link Time Optimization flag, and instructs the plugin compiled here to
+    # use this flag to. This breaks the build for us. Using _fat_ LTO allows
+    # successful linking while still providing LTO benefits. If our build of
+    # `juce` was used as a dependency, we could have patched that `-flto` line
+    # in our juce's source, but that is not possible because it is used as a
+    # Git Submodule.
+    "-ffat-lto-objects"
+  ];
+
   # Fontconfig error: Cannot load default config file: No such file: (null)
   env.FONTCONFIG_FILE = "${fontconfig.out}/etc/fonts/fonts.conf";
-
-  # LTO needs special setup on Linux
-  postPatch = ''
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'juce::juce_recommended_lto_flags' '# Not forcing LTO'
-  '';
 
   installPhase = ''
     runHook preInstall

--- a/pkgs/by-name/sg/sg-323/package.nix
+++ b/pkgs/by-name/sg/sg-323/package.nix
@@ -52,10 +52,16 @@ stdenv.mkDerivation (finalAttrs: {
 
   enableParallelBuilding = true;
 
-  postPatch = ''
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'juce::juce_recommended_lto_flags' '# Not forcing LTO'
-  '';
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isLinux (toString [
+    # juce, compiled in this build as part of a Git submodule, uses `-flto` as
+    # a Link Time Optimization flag, and instructs the plugin compiled here to
+    # use this flag to. This breaks the build for us. Using _fat_ LTO allows
+    # successful linking while still providing LTO benefits. If our build of
+    # `juce` was used as a dependency, we could have patched that `-flto` line
+    # in our juce's source, but that is not possible because it is used as a
+    # Git Submodule.
+    "-ffat-lto-objects"
+  ]);
 
   installPhase =
     let

--- a/pkgs/by-name/so/sonobus/package.nix
+++ b/pkgs/by-name/so/sonobus/package.nix
@@ -76,10 +76,18 @@ stdenv.mkDerivation (finalAttrs: {
   env.NIX_LDFLAGS = lib.optionalString stdenv.hostPlatform.isLinux "-rpath ${lib.makeLibraryPath (finalAttrs.runtimeDependencies)}";
   dontPatchELF = true; # needed or nix will try to optimize the binary by removing "useless" rpath
 
+  env.NIX_CFLAGS_COMPILE = toString [
+    # juce, compiled in this build as part of a Git submodule, uses `-flto` as
+    # a Link Time Optimization flag, and instructs the plugin compiled here to
+    # use this flag to. This breaks the build for us. Using _fat_ LTO allows
+    # successful linking while still providing LTO benefits. If our build of
+    # `juce` was used as a dependency, we could have patched that `-flto` line
+    # in our juce's source, but that is not possible because it is used as a
+    # Git Submodule.
+    "-ffat-lto-objects"
+  ];
+
   postPatch = lib.optionalString (stdenv.hostPlatform.isLinux) ''
-    # needs special setup on Linux, dunno if it can work on Darwin
-    # Also, I get issues with linking without that, not sure why
-    sed -i -e '/juce::juce_recommended_lto_flags/d' CMakeLists.txt
     patchShebangs linux/install.sh
   '';
 

--- a/pkgs/by-name/va/vaporizer2/package.nix
+++ b/pkgs/by-name/va/vaporizer2/package.nix
@@ -31,10 +31,6 @@ stdenv.mkDerivation (finalAttrs: {
   };
 
   postPatch = ''
-    # LTO needs special setup on Linux
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'juce::juce_recommended_lto_flags' '# Not forcing LTO'
-
     rm -rf clap-juce-extensions
     ln -s ${finalAttrs.passthru.clapJuceExtensions} clap-juce-extensions
   '';
@@ -63,6 +59,17 @@ stdenv.mkDerivation (finalAttrs: {
 
   cmakeFlags = [
     (lib.cmakeFeature "USE_SYSTEM_JUCE" "ON")
+  ];
+
+  env.NIX_CFLAGS_COMPILE = toString [
+    # juce, compiled in this build as part of a Git submodule, uses `-flto` as
+    # a Link Time Optimization flag, and instructs the plugin compiled here to
+    # use this flag to. This breaks the build for us. Using _fat_ LTO allows
+    # successful linking while still providing LTO benefits. If our build of
+    # `juce` was used as a dependency, we could have patched that `-flto` line
+    # in our juce's source, but that is not possible because it is used as a
+    # Git Submodule.
+    "-ffat-lto-objects"
   ];
 
   passthru.clapJuceExtensions = fetchFromGitHub {

--- a/pkgs/by-name/zl/zlcompressor/package.nix
+++ b/pkgs/by-name/zl/zlcompressor/package.nix
@@ -75,11 +75,16 @@ clangStdenv.mkDerivation (finalAttrs: {
     "-lXrandr"
   ]);
 
-  # LTO needs special setup on Linux
-  postPatch = lib.optionalString clangStdenv.hostPlatform.isLinux ''
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'juce::juce_recommended_lto_flags' '# Not forcing LTO'
-  '';
+  env.NIX_CFLAGS_COMPILE = lib.optionalString clangStdenv.hostPlatform.isLinux (toString [
+    # juce, compiled in this build as part of a Git submodule, uses `-flto` as
+    # a Link Time Optimization flag, and instructs the plugin compiled here to
+    # use this flag to. This breaks the build for us. Using _fat_ LTO allows
+    # successful linking while still providing LTO benefits. If our build of
+    # `juce` was used as a dependency, we could have patched that `-flto` line
+    # in our juce's source, but that is not possible because it is used as a
+    # Git Submodule.
+    "-ffat-lto-objects"
+  ]);
 
   cmakeFlags = [
     # see: https://github.com/ZL-Audio/ZlEqualizer#clone-and-build

--- a/pkgs/by-name/zl/zlequalizer/package.nix
+++ b/pkgs/by-name/zl/zlequalizer/package.nix
@@ -75,11 +75,16 @@ clangStdenv.mkDerivation (finalAttrs: {
     "-lXrandr"
   ]);
 
-  # LTO needs special setup on Linux
-  postPatch = lib.optionalString clangStdenv.hostPlatform.isLinux ''
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'juce::juce_recommended_lto_flags' '# Not forcing LTO'
-  '';
+  env.NIX_CFLAGS_COMPILE = lib.optionalString clangStdenv.hostPlatform.isLinux (toString [
+    # juce, compiled in this build as part of a Git submodule, uses `-flto` as
+    # a Link Time Optimization flag, and instructs the plugin compiled here to
+    # use this flag to. This breaks the build for us. Using _fat_ LTO allows
+    # successful linking while still providing LTO benefits. If our build of
+    # `juce` was used as a dependency, we could have patched that `-flto` line
+    # in our juce's source, but that is not possible because it is used as a
+    # Git Submodule.
+    "-ffat-lto-objects"
+  ]);
 
   cmakeFlags = [
     # see: https://github.com/ZL-Audio/ZLEqualizer#clone-and-build

--- a/pkgs/by-name/zl/zlsplitter/package.nix
+++ b/pkgs/by-name/zl/zlsplitter/package.nix
@@ -75,11 +75,16 @@ clangStdenv.mkDerivation (finalAttrs: {
     "-lXrandr"
   ]);
 
-  # LTO needs special setup on Linux
-  postPatch = lib.optionalString clangStdenv.hostPlatform.isLinux ''
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'juce::juce_recommended_lto_flags' '# Not forcing LTO'
-  '';
+  env.NIX_CFLAGS_COMPILE = lib.optionalString clangStdenv.hostPlatform.isLinux (toString [
+    # juce, compiled in this build as part of a Git submodule, uses `-flto` as
+    # a Link Time Optimization flag, and instructs the plugin compiled here to
+    # use this flag to. This breaks the build for us. Using _fat_ LTO allows
+    # successful linking while still providing LTO benefits. If our build of
+    # `juce` was used as a dependency, we could have patched that `-flto` line
+    # in our juce's source, but that is not possible because it is used as a
+    # Git Submodule.
+    "-ffat-lto-objects"
+  ]);
 
   cmakeFlags = [
     # see: https://github.com/ZL-Audio/ZlEqualizer#clone-and-build

